### PR TITLE
Create operationOverload

### PR DESCRIPTION
**Pull request by AlekseyVoronkov**

### CODE REVIEW

File: operationOverload [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+            {
+            }
+        }
+
+        public class MatrixIndexException : Exception
+        {
+            public MatrixIndexException(string message) : base(message)
+            {
+            }
+        }
```
Comment (iredb) [inline operationOverload:53]: композиция???

File: operationOverload [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+
+        public static SquareMatrix operator +(SquareMatrix matrix1, SquareMatrix matrix2)
+        {
+            SquareMatrix result = new SquareMatrix(matrix1.matrix.GetLength(0));
```
Comment (iredb) [inline operationOverload:57]: probably matrix1.size

File: operationOverload [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+                    result.matrix[RowIndex, ColumnIndex] = matrix1.matrix[RowIndex, ColumnIndex] + matrix2.matrix[RowIndex, ColumnIndex];
+                }
+            }
+
+            return result;
+        }
+
+        public static SquareMatrix operator *(SquareMatrix matrix1, SquareMatrix matrix2)
+        {
+            SquareMatrix resultMatrix = new SquareMatrix(matrix1.size);
```
Comment (iredb) [inline operationOverload:72]: why do you do another thing here???

File: operationOverload [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+                }
+            }
+
+            return resultMatrix;
+        }
+
+        public static bool operator >(SquareMatrix matrix1, SquareMatrix matrix2)
+        {
+            int count = 0;
+            for (int RowIndex = 0; RowIndex < matrix1.matrix.GetLength(0); ++RowIndex)
```
Comment (iredb) [inline operationOverload:95]: leave a space here, please

Comment (iredb) [reply operationOverload:95]: empty line

File: operationOverload [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+            {
+                for (int ColumnIndex = 0; ColumnIndex < matrix1.matrix.GetLength(1); ++ColumnIndex)
```
Comment (iredb) [inline operationOverload:97]: why first letter is uppercase?

Comment (iredb) [reply operationOverload:97]: remake for all the situations

File: operationOverload [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+                {
+                    if (matrix1.matrix[RowIndex, ColumnIndex] > matrix2.matrix[RowIndex, ColumnIndex])
+                    {
+                        count++;
```
Comment (iredb) [inline operationOverload:101]: oh shit...

File: operationOverload [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+                }
+            }
+            if (count > 2)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
```
Comment (iredb) [inline operationOverload:112]: you can just return count > 2

File: operationOverload [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+        }
+
+        public static bool operator <(SquareMatrix matrix1, SquareMatrix matrix2)
```
Comment (iredb) [inline operationOverload:115]: hmm.. what if matrix1.size more than matrix2.size?

File: operationOverload [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+            }
+            if (count > 1)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
```
Comment (iredb) [inline operationOverload:159]: just do !(matrix1 < matrix2) bro

File: operationOverload [hunk 10]
```diff
@@ -0,0 +1,504 @@
...
+            {
+                for (int ColumnIndex = 0; ColumnIndex < matrix1.matrix.GetLength(1); ++ColumnIndex)
+                {
+                    if (matrix1.matrix[RowIndex, ColumnIndex] > matrix2.matrix[RowIndex, ColumnIndex])
+                    {
+                        count++;
+                    }
+                }
+            }
+            if (count == 2)
```
Comment (iredb) [inline operationOverload:197]: what means 2?

Comment (iredb) [reply operationOverload:197]: make a variable

File: operationOverload [hunk 11]
```diff
@@ -0,0 +1,504 @@
...
+        {
+            return matrix1.size != 1;
+        }
+
+        public static bool operator false(SquareMatrix matrix1)
+        {
+            return matrix1.size == 1;
+        }
+
+        public SquareMatrix GetStepaMatrix(int row, int column)
```
Comment (iredb) [inline operationOverload:250]: :)

File: operationOverload [hunk 12]
```diff
@@ -0,0 +1,504 @@
...
+            {
+                int determinant = matrix[0, 0] * matrix[1, 1] - matrix[1, 0] * matrix[0, 1];
+                return determinant;
+            }
+
+            int result = 0;
+
+            for (int columnIndex = 0; columnIndex < size; ++columnIndex) 
+            {
+
```
Comment (iredb) [inline operationOverload:284]: allo?
where?

File: operationOverload [hunk 13]
```diff
@@ -0,0 +1,504 @@
...
+                {
+                    if (matrix[rowIndex, columnIndex] != other.matrix[rowIndex, columnIndex])
+                    {
+                        return matrix[rowIndex, columnIndex].CompareTo(other.matrix[rowIndex, columnIndex]);
+                    }
+                }
+            }
+
+            return 0; 
+        }
```
Comment (iredb) [inline operationOverload:386]: where is -1?

File: operationOverload [hunk 14]
```diff
@@ -0,0 +1,504 @@
...
+            Console.WriteLine("Second matrix");
+            matrix2.PrintMatrix();
+            Console.WriteLine();
+            Console.WriteLine("There is no such thing as free will, user: ");
+            Console.WriteLine("Make a choice, user: ");
+            Console.WriteLine("1. +; 2. *; 3. <; 4. >; 5. <=; 6. >=;");
+            Console.WriteLine("7. ==; 8. !=; 9. FindDeterminant;");
+            Console.WriteLine("10. FindTransposed; 11. FindInversedMatrix");
+
+
```
Comment (iredb) [inline operationOverload:447]: too deep

File: operationOverload [hunk 15]
```diff
@@ -0,0 +1,504 @@
...
+            };
+            return currentClone;
+        }
+    }
+
+    static class Program
+    {
+        static void Main()
+        {
+            SquareMatrix matrix = new SquareMatrix(3);
```
Comment (iredb) [inline operationOverload:433]: make while(parameter).

File: operationOverload [hunk 16]
```diff
@@ -0,0 +1,504 @@
...
+            SquareMatrix matrix2 = new SquareMatrix(3);
+            Console.WriteLine("First matrix");
+            matrix1.PrintMatrix();
+            Console.WriteLine("Second matrix");
+            matrix2.PrintMatrix();
+            Console.WriteLine();
+            Console.WriteLine("There is no such thing as free will, user: ");
+            Console.WriteLine("Make a choice, user: ");
+            Console.WriteLine("1. +; 2. *; 3. <; 4. >; 5. <=; 6. >=;");
+            Console.WriteLine("7. ==; 8. !=; 9. FindDeterminant;");
```
Comment (iredb) [inline operationOverload:444]: you can use "\n"

**Review by iredb — CHANGES_REQUESTED** [review]

File: operationOverload (hunk 6 above)
Comment (AlekseyVoronkov) [reply operationOverload:101]: oh no...

**Pull request closed without merging**